ocs_ci/ocs/resources/backingstore.py

```diff
@@ -74,33 +81,59 @@ def delete(self):
                 raise e
             pv_name = backingstore_pvc["spec"]["volumeName"]
 
-        if self.method == "oc":
-            OCP(
-                kind="backingstore", namespace=config.ENV_DATA["cluster_namespace"]
-            ).delete(resource_name=self.name)
-        elif self.method == "cli":
-
-            def _cli_deletion_flow():
-                try:
-                    self.mcg_obj.exec_mcg_cmd(f"backingstore delete {self.name}")
+        def _oc_deletion_flow():
+            try:
+                OCP(
+                    kind="backingstore", namespace=config.ENV_DATA["cluster_namespace"]
+                ).delete(resource_name=self.name)
+                return True
+            except CommandFailed as e:
+                if "not found" in e.args[0].lower():
+                    log.warning(f"Namespacestore {self.name} was already deleted.")
                     return True
-                except CommandFailed as e:
-                    if "being used by one or more buckets" in str(e).lower():
+                elif all(
+                    err in e.args[0]
+                    for err in ["cannot complete because pool", "in", "state"]
+                ):
+                    if retry:
```

Comment:
seems strange. Why do we do this for loop?

Reply:
The for is internal to the `all()` creation - it checks that _all_ the strings in the list are also inside the error.
Alternatively, you can imagine it as:
` if all of ["cannot complete because pool", "in", "state"] is in the error message`